File: accepted/100-same-tree.py
# https://leetcode.com/problems/same-tree/
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} p
    # @param {TreeNode} q
    # @return {boolean}
    def isSameTree(self, p, q):
        if None == p or None == q:
            return p == q
        elif p.val != q.val:
            return False
        else:
            return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)

File: accepted/100-same-tree.cs
/* https://leetcode.com/problems/same-tree/ */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        if (null == p || null == q)
            return p == q;
        else if (p.val != q.val)
            return false;
        else
            return IsSameTree(p.left, q.left) && IsSameTree(p.right, q.right);
    }
}
